resources/aws/bucket.go

```diff
@@ -8,13 +8,18 @@ import (
 	"github.com/aws/aws-sdk-go/service/s3"
 	microerror "github.com/giantswarm/microkit/error"
 	"github.com/juju/errgo"
+
+	"github.com/giantswarm/aws-operator/resources"
 )
 
 type Bucket struct {
 	Name string
 	AWSEntity
 }
 
+// Implement ReusableResource.
+var _ = resources.ReusableResource(&Bucket{})
```

Comment:
It's a hack from Kubernetes, they're doing that often.

I had a feeling that we produce a lot of ugly looking code by doing the following in service.go:

```
var foo resources.ResourceXYZ
foo = awsresources.Subnet{}
```

and I think it's better to ensure that struct implements interface in the module it's defined, and just use instances of structs in service.go. But whatever, if it really hurts you that much, I can go back to the old solution.